src/easylink/rule.py

```diff
@@ -110,38 +120,71 @@ class ImplementedRule(Rule):
     """Command to execute."""
     requires_spark: bool
     """Whether or not this ``Implementation`` requires a Spark environment."""
+    is_embarrassingly_parallel: bool = False
+    """Whether or not this ``Implementation`` is to be run in an embarrassingly 
+    parallel way."""
 
-    def _build_rule(self) -> str:
+    def build_rule(self) -> str:
         """Builds the Snakemake rule for this ``Implementation``."""
-        return self._build_io() + self._build_resources() + self._build_shell_command()
+        return self._build_io() + self._build_resources() + self._build_shell_cmd()
 
     def _build_io(self) -> str:
         """Builds the input/output portion of the rule."""
-        return (
+        if self.is_embarrassingly_parallel:
+            # Processed chunks are sent to a 'processed' subdir
+            output_files = [
+                os.path.dirname(file_path)
+                + "/processed/{chunk}/"
+                + os.path.basename(file_path)
+                for file_path in self.output
+            ]
+            log_path_chunk_adder = "-{chunk}"
+        else:
+            output_files = self.output
+            log_path_chunk_adder = ""
+
+        io_str = (
             f"""
 rule:
     name: "{self.name}"
     message: "Running {self.step_name} implementation: {self.implementation_name}" """
             + self._build_input()
             + f"""        
-    output: {self.output}
-    log: "{self.diagnostics_dir}/{self.name}-output.log"
+    output: {output_files}
+    log: "{self.diagnostics_dir}/{self.name}-output{log_path_chunk_adder}.log"
     container: "{self.image_path}" """
         )
+        return io_str
 
     def _build_input(self) -> str:
         input_str = f"""
     input:"""
-        for slot_attrs in self.input_slots.values():
+        input_slots_to_split = self.get_input_slots_to_split(self.input_slots)
+        for slot, attrs in self.input_slots.items():
+            env_var = attrs["env_var"].lower()
+            # FIXME: handle multiple input slots to split
+            if self.is_embarrassingly_parallel and slot == input_slots_to_split[0]:
```

Comment:
Can we make this an error or something for now? I don't like that it looks like this would silently drop index > 0